torch/onnx/utils.py

```diff
@@ -873,51 +878,59 @@ def flatten(x):
         )
 
 
-def _create_jit_graph(model, args):
-    torch_out = None
-    params: Union[List, Tuple]
+def _create_jit_graph(
+    model: torch.nn.Module, args: Sequence[Any]
+) -> Tuple[
+    _C.Graph,
+    List[_C.IValue],
+    Optional[Any],
+    Optional[Union[_C.ScriptModule, _C.ScriptFunction]],
+]:
     if isinstance(model, (torch.jit.ScriptFunction, torch.jit.ScriptModule)):
         flattened_args = tuple(torch.jit._flatten(tuple(args))[0])
         _check_flatten_did_not_remove(args, flattened_args)
-    if isinstance(model, torch.jit.ScriptModule):
-        try:
-            graph = model.forward.graph
-        except AttributeError as e:
-            raise RuntimeError("'forward' method must be a script method") from e
-        _C._jit_pass_onnx_function_substitution(graph)
-        freezed_m = _C._freeze_module(model._c, preserveParameters=True)
-        module, params = _C._jit_onnx_list_model_parameters(freezed_m)
-        method_graph = module._get_method("forward").graph
-        args_params = tuple(args) + tuple(params)
-        param_count_list = _get_param_count_list(method_graph, args_params)
-        in_vars, _ = torch.jit._flatten(args_params)
-        graph = _C._propagate_and_assign_input_shapes(
-            method_graph, tuple(in_vars), param_count_list, False, False
-        )
-        return graph, params, torch_out, module
-    elif isinstance(model, torch.jit.ScriptFunction):
-        params = ()
-        graph = model.graph
-        _C._jit_pass_onnx_function_substitution(graph)
-        param_count_list = _get_param_count_list(graph, args)
-        # FIXME(justinchuby): flattened_args is possibly unbound
-        graph = _C._propagate_and_assign_input_shapes(
-            graph, flattened_args, param_count_list, False, False
-        )
-        return graph, params, torch_out, None
-    else:
-        graph, torch_out = _trace_and_get_graph_from_model(model, args)
-        _C._jit_pass_onnx_lint(graph)
-        state_dict = torch.jit._unique_state_dict(model)
-        params = list(state_dict.values())
-        graph_inputs = list(graph.inputs())
-        user_input_num = len(graph_inputs) - len(state_dict)
-        param_names = list(state_dict.keys())
-        for i, inp in enumerate(graph_inputs):
-            if i >= user_input_num:
-                inp.setDebugName(param_names[i - user_input_num])
-        _C._jit_pass_onnx_function_substitution(graph)
-        return graph, params, torch_out, None
+        torch_out = None
+
+        if isinstance(model, torch.jit.ScriptModule):
+            try:
+                graph = model.forward.graph
+            except AttributeError as e:
+                raise RuntimeError("'forward' method must be a script method") from e
+            _C._jit_pass_onnx_function_substitution(graph)
+            freezed_module = _C._freeze_module(
+                cast(_C.ScriptModule, model._c), preserveParameters=True
```

Comment:
Is this needed due to a missing type annotation on ScriptModule._c?

Reply:
That seems to be the case. Although I am not completely sure why `_c` is needed here?